src/vs/workbench/contrib/comments/browser/commentService.ts

```diff
@@ -307,9 +314,40 @@ export class CommentService extends Disposable implements ICommentService {
 			await this._lastActiveCommentController?.setActiveCommentAndThread(undefined);
 		}
 		this._lastActiveCommentController = commentController;
+		if (commentInfo) {
+			this._activeCommentInfo = { owner: uniqueOwner, thread: commentInfo.thread, comment: commentInfo?.comment };
+		} else {
+			this._activeCommentInfo = undefined;
+		}
 		return commentController.setActiveCommentAndThread(commentInfo);
 	}
 
+	/**
+	 * Navigate to the next or previous comment thread
+	 * @param type
+	 */
+	navigateToCommentAndThread(type: 'next' | 'previous') {
+		const commentInfo = this.activeCommentInfo;
+		if (!commentInfo?.comment || !commentInfo?.thread?.comments) {
+			return;
+		}
+		const currentIndex = this.activeCommentInfo?.thread.comments?.indexOf(commentInfo.comment);
+		if (currentIndex === undefined || currentIndex < 0) {
+			return;
+		}
+		if (type === 'previous' && currentIndex === 0) {
+			return;
+		}
+		if (type === 'next' && currentIndex === commentInfo.thread.comments.length - 1) {
+			return;
+		}
+		const comment = this.activeCommentInfo?.thread.comments?.[type === 'previous' ? currentIndex - 1 : currentIndex + 1];
+		if (!comment) {
+			return;
+		}
+		this.setActiveCommentAndThread(this.activeCommentInfo.owner, { comment, thread: commentInfo.thread });
```

Comment:
This still seems weird to me. `setActiveCommentThread` doesn't change any UI state. Instead, it's supposed to be set inresponse to UI state changing. So when you set it here, the accessible view will change, but the comment that's visible in the viewport won't change. 

Instead of calling `setActiveCommentThread` directly, could we actually reveal the comment? How do other components with accessible views work?

Reply:
If we reveal the comment, the focus change will cause the accessible view to close.

Reply:
We just need the content from the next or previous comment. I was trying to just reuse this function vs adding a bunch, but could add such functions instead?

Reply:
I could add a way to reveal the comment without focusing on it if that doesn't already exist. How do other accessible views work? Do they reveal the "next" and "previous" items without changing focus when the buttons are used?

Reply:
they get the item without changing focus, for example, for chat responses: https://github.com/microsoft/vscode/blob/8ef1efa085057a2d9ab950ebdb9d9d6afd42296b/src/vs/workbench/contrib/chat/browser/chatResponseAccessibleView.ts#L93-L109